lib/src/main/kotlin/RB.kt

```diff
@@ -0,0 +1,526 @@
+package RBTree
+
+import TreeMap.*
+
+enum class Color {
+    RED, BLACK
+}
+
+
+enum class Side {
+    left, right
+}
+
+
+class RBNode<K: Comparable<K>, V>(key: K, value: V, parent: RBNode<K, V>?, color: Color?): Node<K, V, RBNode<K, V>>(key, value, parent){
+    internal var color: Color? = color
+}
+
+
+class RBTree<K: Comparable<K>, V>: TreeMap<K, V, RBNode<K, V>>() {
+    override var root: RBNode<K, V>? = null
+
+    // поворот налево: n - новая "старшая вершина", отец n становится левым ребенком n
+    private fun leftRotation(n: RBNode<K, V>?) {
+        var updateRoot = false
+        if (n?.parent == this.root) {
+            updateRoot  = true
+        }
+
+        if (n == null) {
+            return
+        }
+        var parent = n.parent
+        if (parent == null) {
+            return
+        }
+        parent.rightChild = n.leftChild
+        if (parent.rightChild != null) {
+            parent.rightChild?.parent = parent
+        }
+        n.leftChild = parent
+        n.parent = parent.parent
+        if (parent.parent != null) {
+            if (parent == parent.parent?.leftChild) {
+                parent.parent?.leftChild = n
+            }
+            else {
+                parent.parent?.rightChild = n
+            }
+        }
+        parent.parent = n
+
+        if (updateRoot) {
+            this.root = n
+        }
```

Comment:
Можно было обойтись без updateRoot
```suggestion
         if (n.parent == this.root) {
            this.root = n
        }

        if (n == null) {
            return
        }
        var parent = n.parent
        if (parent == null) {
            return
        }
        parent.rightChild = n.leftChild
        if (parent.rightChild != null) {
            parent.rightChild?.parent = parent
        }
        n.leftChild = parent
        n.parent = parent.parent
        if (parent.parent != null) {
            if (parent == parent.parent?.leftChild) {
                parent.parent?.leftChild = n
            }
            else {
                parent.parent?.rightChild = n
            }
        }
        parent.parent = n
```